android/src/main/java/com/bitmovin/player/reactnative/AnalyticsModule.kt

```diff
@@ -126,6 +126,17 @@ class AnalyticsModule(private val context: ReactApplicationContext) : ReactConte
         }
     }
 
+    @ReactMethod
+    fun addSourceMetadata(nativeId: NativeId, playerId: NativeId?, json: ReadableMap?) {
+        uiManager()?.addUIBlock { _ ->
+            playerModule()?.getPlayer(playerId)?.source?.let { playerSource ->
+                JsonConverter.toAnalyticsSourceMetadata(json)?.let { sourceMetadata ->
+                    collectors[nativeId]?.addSourceMetadata(playerSource, sourceMetadata)
+                }
+            }
```

Comment:
Let's make this a bit easier to digest by removing one layer please.

```suggestion
            val playerSource = playerModule()?.getPlayer(playerId)?.source ?: return
            JsonConverter.toAnalyticsSourceMetadata(json)?.let { sourceMetadata ->
                collectors[nativeId]?.addSourceMetadata(playerSource, sourceMetadata)
            }
```

Reply:
Thanks for the suggestion!
Taken!